site/en/blog/how-construct3-uses-the-file-system-access-api/index.md

```diff
@@ -0,0 +1,70 @@
+
+[Construct 3](https://www.construct.net/en/make-games/free-trial) is a game editor developed by the brothers [Thomas and Ashley Gullen](https://www.starterstory.com/stories/we-created-a-100k-month-no-code-tool-for-video-game-developers). For the present third iteration of their game editor, the two fully _"[bet] on the browser being the new operating system"_ after building for Windows and [NW.js](https://nwjs.io/) before. You can get an impression of some games developed with the editor by exploring its [showcase](https://www.construct.net/en/make-games/showcase) or working through the [guided tour](https://editor.construct.net/?startTour). Thanks to the superpowers of the web, you can also just click through to any of the ["Get inspired" examples](https://www.construct.net/en/make-games/free-trial#:~:text=the%20Guided%20Tour-,Get%20inspired,-OPEN%20AND%20EDIT) and start editing immediately.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/GYwJISahyPnsfSgsX3N1.png", alt="The Construct 3 app showing the user editing one of the starter projects.", width="800", height="522" %}
+
+## The File System Access API in Construct 3
+
+Construct provides the option of saving to local files with the [File System Access API](https://developer.mozilla.org/docs/Web/API/File_System_Access_API), as well as cloud save (Google Drive, OneDrive, Dropbox), and downloading of a copy of the project file. Stats the Construct developers have collected show that 65% of saves are done with the File System Access API, which demonstrates it's what most customers want to use.
+
+For saving, the following snippet shows the original production code for obtaining a [`FileSystemFileHandle`](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle) from the [`showSaveFilePicker()`](https://developer.mozilla.org/docs/Web/API/Window/showSaveFilePicker) method, which then serves to save the actual data. Construct makes use of the `id` options parameter.  The `id` field can be specified to suggest the directory in which the file picker opens. By specifying an `id`, the browser can remember different directories for different IDs, which serves to start the dialog consistently in the same directory depending on the ID. For example, level files could open in `Documents/levels/`, whereas texture files could open in `Images/textures/`.  The `types` parameter is an array of supported file types with a localized user-visual `description` and an `accept` object that tells the operating system to initially only accept `.c3p` files with the MIME type `application/x-construct3-project`.
+
+```js
+let fileHandle = null;
+try {
+  fileHandle = await window["showSaveFilePicker"]({
+    "id": "save-project-file",
+    "types": [{
+	 "description":
+          lang("ui.project-file-picker.c3-single-file-project"),
+      "accept": {
+	   "application/x-construct3-project": [".c3p"]
+ }
+    }]
+  });
+} catch (err) {
+  // Assume user cancelled, or permission otherwise denied.
+  return;
+}
```

Comment:
```suggestion
let fileHandle = null;
try {
  fileHandle = await window["showSaveFilePicker"]({
    id: "save-project-file",
    types: [
      {
        description: lang("ui.project-file-picker.c3-single-file-project"),
        accept: {
          "application/x-construct3-project": [".c3p"],
        },
      },
    ],
  });
} catch (err) {
  // Assume user cancelled, or permission otherwise denied.
  return;
}
```